expertsystem/state/particle.py

```diff
@@ -258,6 +264,198 @@ def convert_to_dict(
+            f"initial_state={self.initial_state}, "
+            f"final_state={self.final_state})"
+        )
+
+    def __contains__(self, other: object) -> bool:
+        """Check if a `KinematicRepresentation` is contained within another.
+
+        You can also compare with a `list` of `list` instances, such as:
+
+        .. code-block::
+
+            [["gamma", "pi0"], ["gamma", "pi0", "pi0"]]
+
+        This list will be compared **only** with the
+        `~KinematicRepresentation.final_state`!
+        """
+
+        def is_sublist(
+            sub_representation: Optional[List[List[Any]]],
+            main_representation: Optional[List[List[Any]]],
+        ) -> bool:
+            if main_representation is None:
+                if sub_representation is None:
+                    return True
+                return False
+            if sub_representation is None:
+                return True
+            for group in sub_representation:
+                if group not in main_representation:
+                    return False
+            return True
+
+        if isinstance(other, KinematicRepresentation):
+            return is_sublist(
+                other.initial_state, self.initial_state
+            ) and is_sublist(other.final_state, self.final_state)
+        if isinstance(other, list):
+            for item in other:
+                if not isinstance(item, list):
+                    raise ValueError(
+                        "Comparison representation needs to be a list of lists"
+                    )
+            return is_sublist(other, self.final_state)
+        raise ValueError(
+            f"Cannot compare {self.__class__.__name__} with {other.__class__.__name__}"
+        )
+
+    def __import(
+        self, nested_list: Union[Sequence[Sequence[Any]], Sequence[Any]]
+    ) -> List[List[Any]]:
+        return self.__sort(self.__prepare(nested_list))
+
+    def __prepare(
+        self, nested_list: Union[Sequence[Sequence[Any]], Sequence[Any]]
+    ) -> List[List[Any]]:
+        if len(nested_list) == 0 or not isinstance(nested_list[0], list):
+            nested_list = [nested_list]
+        return [
+            [self.__extract_particle_name(item) for item in sub_list]
+            for sub_list in nested_list
+        ]
+
+    @staticmethod
+    def __sort(nested_list: List[List[Any]]) -> List[List[Any]]:
+        return sorted([sorted(sub_list) for sub_list in nested_list])
+
+    @staticmethod
+    def __extract_particle_name(item: object) -> str:
+        if isinstance(item, str):
+            return item
+        if isinstance(item, (tuple, list)) and isinstance(item[0], str):
+            return item[0]
+        if isinstance(item, Particle):
+            return item.name
+        if isinstance(item, dict) and "Name" in item:
+            return str(item["Name"])
+        raise ValueError(
+            f"Cannot extract particle name from {item.__class__.__name__}"
+        )
```

Comment:
This can maybe be simplified when combining it with `get_kinematic_representation()`

Reply:
https://github.com/ComPWA/expertsystem/pull/245#discussion_r491052122